mesonbuild/modules/__init__.py

```diff
@@ -112,7 +112,12 @@ def find_tool(self, name: str, depname: str, varname: str, required: bool = True
         if dep.found() and dep.type_name == 'pkgconfig':
             value = dep.get_variable(pkgconfig=varname)
             if value:
-                return ExternalProgram(value)
+                progobj = ExternalProgram(value)
+                if not progobj.found():
+                    msg = (f'Dependency {depname!r} tool variable {varname!r} contains erroneous value: {value!r}\n\n'
+                           'This is a distributor issue -- please report it to your {depname} provider.')
+                    raise mesonlib.MesonException(msg)
```

Comment:
Should we raise or print warning? Maybe downgrade now and make it fatal later since we had people hitting this case?

Reply:
The issue is that beforehand, as far as I understand, these people were getting failing `ninja` since we blindly assumed the path in the pkg-config file worked. Then we refactored to fix the .exe case and the side effect was any *other* not-found cases would hit a python logic error and fail with an unhandled exception.

Reply:
Ok, let's do that, worst case we can downgrade in 1.3.1 if someone complains.